api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -15,31 +15,62 @@
  */
 package io.cloudevents;
 
+import com.fasterxml.jackson.databind.JsonNode;
+
 import java.util.Map;
 import java.util.Optional;
 
 /**
  * An abstract event envelope
- * @param <A> The attributes type
- * @param <T> The 'data' type
  * @author fabiojose
+ * @author slinkydeveloper
  */
-public interface CloudEvent<A extends Attributes, T> {
+public interface CloudEvent {
 
     /**
      * The event context attributes
      */
-    A getAttributes();
+    Attributes getAttributes();
 
     /**
      * The event data
      */
-    Optional<T> getData();
+    Optional<String> getDataAsString() throws DataConversionException;
 
-    byte[] getDataBase64();
+    /**
+     * The event data
+     */
+    Optional<byte[]> getDataAsBytes() throws DataConversionException;
+
+    /**
+     * The event data
+     */
+    Optional<JsonNode> getDataAsJson() throws DataConversionException;
 
     /**
      * The event extensions
+     *
+     * Extensions values could be String/Number/Boolean
      */
     Map<String, Object> getExtensions();
+
+    /**
+     * Write an extension into this cloud event
+     * @param e
+     */
+    default void writeExtension(Extension e) {
+        e.writeToEvent(this);
+    }
+
+    static io.cloudevents.v1.CloudEventBuilder build() {
```

Comment:
this method should either return a common, non-versioned `CloudEventBuilder` interface, or be removed.

Otherwise, if for example spec `1.1` is released, changing this method to `io.cloudevents.v1_1.CloudEventBuilder build` would be a binary breaking change, and I expect it to be the most used way of creating the event.